fix: Chart widget (#1022)
appsmithorg/appsmith-docs: website/docs/reference/widgets/chart.mdx

Appsmith offers the ability to display both static and dynamic data in a chart. The data should be provided in a JSON format:

```js
[
  {
    "x": "Product1",
    "y": 20000
  },..
]
```

In this format, **`x`** denotes the **label** and **`y`** denotes the **value**. It's important to provide a value when displaying data to ensure that the chart can render correctly. Please note that the JSON format provided is only suitable for basic charts

Appsmith offers several quick and easy ways to represent your data. With the Chart Type property, you can select one of the following types to visualize your data:
* Line Charts, 
* Bar Charts, 
* Pie Charts, 
* Column Charts, 
* Area Charts, and  
* [Custom Charts](#custom-charts)

### Display static data

To display static data, you can enter the desired data directly into the `Series Data property`, and it remains the same until manually updated or edited. This method is useful when you need to display fixed information, such as trends or comparisons, that don't change based on user interactions. 

```js
[
  {
    "x": "2021",
    "y": 200000
  },
  {
    "x": "2022",
    "y": 320000
  }
]
```

### Display data dynamically

> You can display dynamic data by fetching it from APIs or databases. This can be done by using binding values to **Series Data property**. If you need to transform the incoming data to match this format, you can use **JSObject** to perform the transformation before binding it to the chart widget.

Reviewer: ```suggestion
You can display dynamic data from queries or JS functions by binding the response to the **Series Data** property. If you need to transform the incoming data to match the expected structure, you can use JavaScript code to perform the transformation before binding it to the Chart widget.
```

For example, lets say you have a PostgreSQL database containing information about assets in your organization, such as their types and the date they were purchased. You can create a query that groups the assets by type and counts how many there are of each type, like this:

> ```sql

Reviewer: this example shows the transformation done in the query itself.  

so move the JS line below this example and give an example of the mustache syntax

select type as x, count (*) as y from asset group by type
```

This query returns data in the required JSON format, where `x` represents the asset type and `y` represents the count of assets of that type. Next, in the chart widget's **Series** property, add `{{assetquery.data}}` to bind the data returned by the query to the chart. This displays a dynamic chart that updates as the data in the database changes.

> ### Display multiple data columns

Reviewer: might need to think of a different word other than columns

If you want to display multiple data columns on a chart, you can do so by using the **Add Series** property. This feature allows you to populate the chart with different sets of data and customize various aspects of its appearance, such as colors and labels, to create a visually appealing and informative chart.

For example, if you have data on the number of males and females who voted in each year of a political election, you can use the Chart Series property to display both sets of data on the same chart. This would allow you to compare the number of male and female voters over time.

> Check this [sample app](https://app.appsmith.com/applications/620b668f6b4b1e154a3c2418/pages/620b8bc56b4b1e154a3c2688) to learn how to produce a dual-axis chart combining columns, lines, and areas.

Reviewer: this sample is for custom charts. Let's give an example. even a screenshot showing this example will do

> ## Perform actions based on data point selection

Reviewer: ```suggestion
## Trigger actions on data point selection
```

> 

Reviewer: We need to explain with a a proper usecase

Performing actions based on data point selection refers to the ability to execute certain actions or operations when a user clicks on a specific data point in a chart widget. 

To achieve this, you can set up the **"onDataPointClick"** event to perform a designated action. For instance, you can use a Modal widget to show the details of the clicked data point using `{{Chart1.selectedDataPoint.x}}` or you can execute a query based on the user's selection.

## Custom charts

> Appsmith offers integration with [FusionCharts](https://www.fusioncharts.com/dev/chart-guide/list-of-charts/) and provides a wide range of chart types that can be used to create custom charts. Once you have selected the Custom Chart type, you can configure the FusionCharts using the Custom Fusion Chart property.

Reviewer: ```suggestion
Appsmith offers integration with [FusionCharts](https://www.fusioncharts.com/dev/chart-guide/list-of-charts/) and provides a wide range of chart types that can be used to create custom charts. Once you have selected the **Chart Type > Custom Chart**, you can configure FusionCharts using the **Custom Fusion Chart** property.
```

:::info
FusionCharts provides an extensive range of chart variants, with almost 100+ options to choose from. To learn more about these charts, you can refer to the official [FusionCharts documentation](https://www.fusioncharts.com/dev/chart-guide/list-of-charts/) 
:::

> To create a custom chart, you need to provide two keys in your configuration: type and datasource:

Reviewer: mention this is a part of the Custom Fusion Chart property

* **Type**

The type is used to define the chart type recognized by the FusionCharts. You can[ get a list of all the supported types available](https://www.fusioncharts.com/dev/chart-guide/list-of-charts) on FusionCharts.

* **Datasource**

The datasource defines the customization options and the data points to create a chart. It includes two attributes: **chart** and **data**.

The chart object can customize the chart and includes options like caption, xAxisName, and yAxisName. On the other hand, the data attribute represents the data in an array format with key-value pairs like `[{“label”: “string value”, “value”: “string value”}]`.

Below is an example of a configuration for the Custom Fusion Chart property:

```
{
  "type": "",
  "dataSource": {
    "chart": {},
    "data": []
  }
}
```

> ### Custom charts examples

Reviewer: Just Examples

Here are some examples of custom charts that you can use as a reference to create your own customized charts.

> <Tabs>

Reviewer: in the custom chart example give the entire json code so that the user can copy paste to test the example

| **x-axis Label**             	| String    	| Sets the text which appears as a label for the x-axis.                                                                                                                                                                        	|
| **y-axis Label**             	| String    	| Sets the text which appears as a label for the y-axis.                                                                                                                                                                        	|
| **x-axis Label Orientation** 	| String    	| Sets the size/rotation behavior for the x-axis label text. Chose from Auto, Slant, Rotate, or Stagger.                                                                                                                        	|
| **Adaptive Axis**            	| Boolean   	| Determines the scaling behavior of the y-axis. OFF: The y-axis begins at zero and spans to an upper limit based on the data points; ON: The y-axis starting and ending values are both determined based upon the data points. 	|

> ### Reference properties

Reviewer: add chartData
